codegen/src/codegen.ts

```diff
@@ -123,10 +123,10 @@ export async function generate({
     export class XataClient extends BaseClient<{
       ${tables.map((table) => `"${table.name}": Repository<${getTypeName(table.name)}>;`).join('\n')}
     }> {
-      constructor(options: XataClientOptions) {
+      constructor(options?: Partial<XataClientOptions>) {
```

Comment:
```suggestion
      constructor(options?: XataClientOptions) {
```

Shouldn't all properties be now optional?

Reply:
The fields `branch` and `apiKey` are required in our internal code. The user may or may not specify them, but we will calculate values for them. So, either we put all as optional and internally we have a different type or we use `options?.apiKey` instead of `options.apiKey`.

Reply:
Yeah, don't see why we can't make everything optional. Proposal: https://github.com/xataio/client-ts/pull/165